PlagiarismPlugin.inc.php

```diff
@@ -43,215 +106,905 @@ public function getDescription() {
 	/**
 	 * @copydoc LazyLoadPlugin::getCanEnable()
 	 */
-	function getCanEnable($contextId = null) {
+	public function getCanEnable($contextId = null) {
 		return !Config::getVar('ithenticate', 'ithenticate');
 	}
 
 	/**
 	 * @copydoc LazyLoadPlugin::getCanDisable()
 	 */
-	function getCanDisable($contextId = null) {
+	public function getCanDisable($contextId = null) {
 		return !Config::getVar('ithenticate', 'ithenticate');
 	}
 
 	/**
 	 * @copydoc LazyLoadPlugin::getEnabled()
 	 */
-	function getEnabled($contextId = null) {
+	public function getEnabled($contextId = null) {
 		return parent::getEnabled($contextId) || Config::getVar('ithenticate', 'ithenticate');
 	}
 
 	/**
-	 * Fetch credentials from config.inc.php, if available
-	 * @return array username and password, or null(s)
-	**/
-	function getForcedCredentials() {
+	 * Add properties for this type of public identifier to the submission entity's list for
+	 * storage in the database.
+	 * 
+	 * @param string $hookName `Schema::get::submission`
+	 * @param array $params
+	 * 
+	 * @return bool
+	 */
+	public function addPlagiarismCheckDataToSubmissionSchema($hookName, $params) {
+		$schema =& $params[0];
+
+		$schema->properties->ithenticate_eula_version = (object) [
+			'type' => 'string',
+			'description' => 'The iThenticate EULA version which has been agreed at submission checklist',
+			'apiSummary' => true,
+			'validation' => ['nullable'],
+		];
+
+		$schema->properties->ithenticate_eula_url = (object) [
+			'type' => 'string',
+			'description' => 'The iThenticate EULA url which has been agreen at submission checklist',
+			'apiSummary' => true,
+			'validation' => ['nullable'],
+		];
+
+		$schema->properties->ithenticate_submission_completed_at = (object) [
+			'type' => 'string',
+			'description' => 'The timestamp at which this submission successfully completed uploading all files at iThenticate service end',
+			'apiSummary' => true,
+			'validation' => [
+				'date:Y-m-d H:i:s',
+				'nullable',
+			],
+		];
+
+		return false;
+	}
+
+	/**
+	 * Add properties for this type of public identifier to the submission file entity's list for
+	 * storage in the database.
+	 * 
+	 * @param string $hookName `Schema::get::submissionFile`
+	 * @param array $params
+	 * 
+	 * @return bool
+	 */
+	public function addPlagiarismCheckDataToSubmissionFileSchema($hookName, $params) {
+		$schema =& $params[0];
+
+		$schema->properties->ithenticate_id = (object) [
+			'type' => 'string',
+			'description' => 'The iThenticate submission id for submission file',
+			'apiSummary' => true,
```

Comment:
I think pretty much everything this plugin adds shouldn't be added to the API summary.